### ISSUE TITLE
Drive doesn't delete files the account doesn't own

### ISSUE THREAD

**Issue by ChrisSLT** [ncw/rclone#34]
1.10 is not deleting files that are not owned by the account owner.  This gets in the way of keeping shared folders in sync.

Through the web-based UI you can still delete non-owned files, and you can also do it through Google Apps Scripts, so I'm pretty sure there is an API call you can make that will allow it.  Try removeFile - that worked in an Apps Script I tried (and some other API calls to send the file to trash did *not* work).

Thank you

OK, more info:
1) When you purge a non-empty directory, even if that directory contains non-owned files, it works. But the problem is the only way to do it now is to remove the whole directory, and if it's a share directory, then you have to create a new directory and reshare it which isn't a good workflow.

2) When you run: rclone purge remote:SharedFolder/nonOwnedFile.txt it gives the errors:
-Error 403, insufficient permissions.  This is incorrect, as the owner of the folder does have permission to delete a shared file.
-Failed to purge: can't rmdir in limited fs

3) When you run rclone purge remote:SharedFolder/OwnedFile.txt it actually does delete the file but it gives the error "Failed to purge: can't rmdir in limited fs."

**Comment by ncw** [comment]
How do you make a non-owned file?  I'll have a go at reproducing your problem it if I can do that!

Thanks

Nick

**Comment by ChrisSLT** [comment]
Thanks again for the quick response.

If you can make it possible to move files (delete a file after verification of successful copy) for remote to local and local to remote, that would really help us in our environment, and make this issue here less important!

**Comment by ChrisSLT** [comment]
But to answer your question, you create a folder in drive under the account that is in the config file.  then share that folder with someone else, and that person uploads files into the shared folder.  That's it.  Rclone is having trouble deleting files that were uploaded into the shared folder by someone other than the account owner

**Comment by ncw** [comment]
I could [trash](https://developers.google.com/drive/v2/reference/files/trash) these files instead of deleting them - you can trash a file you don't own. What do you think of that idea?

**Comment by ChrisSLT** [comment]
Great! Can you make it possible to trash individual files and a folder?

**Comment by ncw** [comment]
The API docs don't say if you can trash folders, but I expect that will work as it works with delete.

**Comment by ChrisSLT** [comment]
I think rclone already purges folders, regardless of ownership, and that it
works, even though it throws an error

I'm talking only of Google Drive

On Wed, Apr 22, 2015 at 10:07 AM, Nick Craig-Wood <notifications@github.com>
wrote:

> The API docs don't say if you can trash folders, but I expect that will
> work as it works with delete.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ncw/rclone/issues/34#issuecomment-95192153>.
>



-- 
347.708.6549

**Comment by ncw** [comment]
Maybe I'm not understanding your request?

> Can you make it possible to trash individual files and a folder?

**Comment by ChrisSLT** [comment]
Sorry - I mean, can you make it possible to:

1) Trash or delete an individual file

2) Trash or delete an individual folder and all the files within it (even it's not empty)

**Comment by ncw** [comment]
I see what you mean.

Just talking about delete for the moment.

For 1) this should work but doesn't :-(

    rclone -v purge drive:testdel/a.txt

For 2) purge should do the trick

    rclone -v purge drive:testdel

As for choosing whether to `trash` or `delete` I could
  * try deleting first, then if that fails, try trashing - that would fix your initial problem
  * make a config file option which affects the behaviour of purge / and deleting files (eg when running `sync`)
  * make a command line option to do the same
  * make a new command `trash` which works like `purge` otherwise
    * this doesn't help with the `sync` command though, unless I make a `sync_and_trash` option which is starting to sound messy

Which would you prefer?

**Comment by ChrisSLT** [comment]
Great to have you back!  Was worried we'd lost you for a little while.

OK, let me back up and try to be clearer.  Here's what I think would be really helpful:

1) Be able to MOVE files.  This is the most important.  Rclone would copy a file and then after it checked to be sure the file copied correctly, it would delete it from the source.

2) Deleting an individual file or folder is important but secondary

**Comment by ncw** [comment]
I was away - but I'm back now!

The move command is in progress in #35 so I don't want to discuss that on this ticket!

I'd like to figure out `delete` vs `trash` as I think it would be helpful for drive users.

**Comment by ChrisSLT** [comment]
Oh - great news on the move command.  OK.

Your idea to allow delete vs trash sounds fine to me

**Comment by ncw** [comment]
OK will experiment - watch this space!

**Comment by ChrisSLT** [comment]
Hi Nick,
 Can you tell me where you are with regard with being able to 

(1) delete (or send to trash) an individual file, or all files in a folder, in Drive?

(2) move an individual file, or all files in a folder, in Drive?

Thank you
Chris

**Comment by ncw** [comment]
1) I made an issue for that #82 

2) I have a prototype in a branch which I need to test and merge

Both waiting on a stretch of free time!

**Comment by ChrisSLT** [comment]
Hi Nick - do you have time/bandwidth/interest in working on this?  Ultimately, the goal is to:

1) Be able to send an individual file, and a folder, in Drive to trash

2) Be able to move an individual file, or a folder, in Drive (this would mean it's copied to a new location and then after verification of successful copy is deleted from current location)

I can help test etc. but unfortunately can't do much coding in Go..

Thank you,
Chris

**Comment by ncw** [comment]
@ChrisSLT Sorry for the lack of progress on this!  I've been away on vacation and I have rather a lot of other commitments.  I'd particularly like to finish 1) as it has been in a branch for ages, and 2) is pretty easy, so don't despair!

**Comment by ChrisSLT** [comment]
Thanks!

So hopefully we're looking at the following working, right?

1) MOVE
rclone mv Drive:folder/ /Volumes/Mac/User/Desktop/
rclone mv Drive:folder/file.txt /Volumes/Mac/User/Desktop/

2) DELETE
rclone rmdir Drive:/folder
rclone del Drive:/folder/file.txt

It's OK and probably preferable to send this stuff to trash!

**Comment by ChrisSLT** [comment]
Hi Nick,
 I know you've been doing some work on deleting and moving.  I just tried

rclone purge Drive:file.txt

and while it does delete the file, it actually tells you "Failed to purge: Can't rmdir in limited fs."

Just heads up

thanks

**Comment by ncw** [comment]
@ChrisSLT thanks for that.  I've put that in a new issue #112

Have you tried the `--drive-use-trash` new in rclone 1.18 which should fix the original issue in this ticket.  This works for files and directories.

**Comment by ncw** [comment]
I think all of these issues are fixed now.

  * `--drive-use-trash` should fix the can't delete non owned files #82
  * `rclone move` has been implemented #35
  * `rclone purge` is now fixed for files in #112

If I've missed anything then please open a new issue as this one is too long now!

Thanks

Nick

**Comment by ChrisSLT** [comment]
I'll try to test this soon.  thanks for the amazing work these past few months.

**Comment by memeplex** [comment]
Is it possible to configure `--drive-use-trash` as the default mode? Currently I'm using a bash alias, but I'm still a bit paranoid (having lost some files after a no use-trash sync). I believe rclone conf is limited to configure the stores, but maybe I'm wrong.

**Comment by ncw** [comment]
@memeplex you are correct there isn't a way of putting this in the config file at the moment.  Can you make a new issue with your request please?

Thanks

Nick

**Comment by memeplex** [comment]
@ncw: https://github.com/ncw/rclone/issues/554
